x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/detection_rules_client/converters/common_params_camel_to_snake.ts

```diff
@@ -9,6 +9,9 @@ import { convertObjectKeysToSnakeCase } from '../../../../../../utils/object_cas
 import type { BaseRuleParams } from '../../../../rule_schema';
 import { migrateLegacyInvestigationFields } from '../../../utils/utils';
 
+/**
+ * @deprecated Use convertObjectKeysToSnakeCase instead
+ */
```

Comment:
I see your point about naming/responsibilities, and I can add a comment about it migrating investigation fields as well. This change came from @xcrzx's suggested patch: 17f68a39d28f217b59df5f9c867b94702af7dcb0, so he might have more insight here.

Reply:
The initial idea was that case converters shouldn’t handle any kind of data migration. So instead of this function, we can use the generic `convertObjectKeysToSnakeCase` and move the migration logic to a more appropriate place, perhaps where we normalize the data or something similar. 

Once all the main client methods are implemented, it would be a good idea to revisit the `RulesManagementClient` and figure out how to best split up the different responsibilities.